engine/src/org/pentaho/di/trans/steps/execprocess/ExecProcess.java

```diff
@@ -110,7 +128,21 @@ public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws
       }
 
       // execute and return result
-      execProcess( processString, processResult );
+      if ( meta.isArgumentsInFields() ) {
+        List<String> cmdArray = new ArrayList<>();
+        cmdArray.add( processString );
+
+        for ( int i = 0; i < data.indexOfArguments.length; i++ ) {
+          String argString = data.previousRowMeta.getString( r, data.indexOfArguments[i] );
+          if ( !Const.isEmpty( argString ) ) {
```

Comment:
Why restrict them from having empty arguments ? Seems that people do this often enough ( e.g. PDI-12510 ).